.github/workflows/snapshot.yaml

```diff
@@ -0,0 +1,43 @@
+---
+
+name: Publish Snapshot
+
+on:
+  push:
+    branches:
+      - main
+      - '[0-9].x'
+  workflow_dispatch:
+
+jobs:
+
+  snapshot:
+    name: Deploy Snapshot
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up publishing to maven central
+        uses: actions/setup-java@v4
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: Cache dependencies
```

Comment:
Not that important:
We can skip this entire step if we use the built-in caching of the setup-java step (the previous one)
See cicd.yaml

Reply:
Thanks for the suggestion! Done.